### PR TITLE
chore: remove `bytebuffer` dependency

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description

Based on https://github.com/trezor/trezor-suite/pull/10294

### CODE REVIEW

File: packages/protobuf/tests/encode-decode-basic.test.ts [hunk 1]
```diff
@@ -231,7 +231,8 @@ describe('basic concepts', () => {
             });
         });
 
-        test('Different protobuf between receiving ends', () => {
+        // TODO: what does it test exactly @mroz22
+        test.skip('Different protobuf between receiving ends', () => {
```
Comment (mroz22) [inline packages/protobuf/tests/encode-decode-basic.test.ts]: imho there was no real usecase for it, just curiosity "what will happen".

File: packages/protocol/src/protocol-bridge/encode.ts [hunk 2]
```diff
@@ -1,29 +1,24 @@
...
-    encodedByteBuffer.writeUint32(data.limit);
+    encodedBuffer.writeUInt32BE(data.length, 2);
 
     // then put in the actual message
-    encodedByteBuffer.append(data.buffer);
-
-    encodedByteBuffer.reset();
+    data.copy(encodedBuffer, HEADER_SIZE);
 
     // todo: it would be nicer to return Buffer instead of ByteBuffer. The problem is that ByteBuffer.Buffer.toString behaves differently in web and node.
```
Comment (mroz22) [inline packages/protocol/src/protocol-bridge/encode.ts]: nitpick: comment could go away :P

**Review by mroz22 — APPROVED** [review]
looks good!

File: packages/protocol/src/protocol-bridge/encode.ts (hunk 2 above)
Comment (szymonlesisz) [reply packages/protocol/src/protocol-bridge/encode.ts]: https://github.com/trezor/trezor-suite/pull/10305/commits/49528f0e0825c830d88172f047073fb7e14ee671

**Pull request merged**